### ISSUE TITLE
[Compile Issue] Statics are not hoisted correctly

### ISSUE THREAD

**Issue by tinkerer-red** [YoYoGames/GameMaker-Bugs#2211]
### Description

Static declarations will only be hoisted to the top most static, and not to the top of the function block.

The following code will fail.
```gml
function foo() {
    if (false) {
        static _fake = undefined
    }
    static bar = function(){show_debug_message("Was hoisted correctly")};
    
    bar();
}
//run it
foo()
```

Further evident they are not hoisted correctly, you can see the non-manually hoisted static actually takes longer:
![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/48543299/392b1ca0-e95d-473d-82d3-cfd7cc23b5a2)
![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/48543299/2307624b-c6d5-41e9-8301-5f4bc1490300)

### Expected Change

statics should be hoisted to the top of the function block.

### Steps To Reproduce

See description for code

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2023.8.2 (Monthly)

### Which platform(s) are you seeing the problem on?

Windows

### Contact Us Package Attached?

- [X] I have attached my Contact Us Package

### Sample Project Added?

- [X] I have included a small sample project

**Comment by rwkay** [comment]
This was fixed in 2023.11 last week with https://github.com/YoYoGames/GameMaker/commit/c7612c33f0da7dfc22c4bcc2d00d463068dca63b

**Comment by caitlinrmcintyre** [comment]
Verified in IDE v2023.1100.0.450  Runtime v2023.1100.0.459, thanks.
